lib/brexit_checker/action.rb

```diff
@@ -2,19 +2,23 @@ class BrexitChecker::Action
   include ActiveModel::Validations
 
   CONFIG_PATH = Rails.root.join("lib", "brexit_checker", "actions.yaml")
+  GROUPS_PATH = Rails.root.join("lib", "brexit_checker", "groups.yaml")
 
   validates_presence_of :id, :title, :consequence, :criteria
   validates_inclusion_of :audience, in: %w(business citizen)
   validates_presence_of :guidance_link_text, if: :guidance_url
   validates_numericality_of :priority, only_integer: true
+  validate :citizen_actions_must_have_groupings
 
   attr_reader :id, :title, :consequence, :exception, :title_url,
               :lead_time, :criteria, :audience, :guidance_link_text,
-              :guidance_url, :guidance_prompt, :priority
+              :guidance_url, :guidance_prompt, :priority, :result_groups,
+              :groups
 
   def initialize(attrs)
     attrs.each { |key, value| instance_variable_set("@#{key}", value) }
     validate!
+    load_groups
```

Comment:
I'm not sure about merging in the groups here. 

This is an action and should not really know about groups. It is also a bit confusing to have a variable called 'groups' and 'result_groups'. It is not immediately clear which-is-which

Reply:
So i think my theory was, if I could load groups here.
Then each action goes from having just a `results_group` array of strings, to an array of proper rich groups, which have a priority key.

Allows me to do this here:
[all_groups = actions.map(&:groups).flatten.uniq](https://github.com/alphagov/finder-frontend/blob/98ce7c3efecb738b4782ee11b2d456b43af46111/app/helpers/brexit_checker_helper.rb#L66)